### ISSUE TITLE
Global styles for buttons and heading are not reflecting

### ISSUE THREAD

**Issue by kspilarski** [Automattic/wp-calypso#68916]
### Quick summary

Using Zoologist and Twenty Twenty Two red theme, setting global styles to buttons (like size, appearance and colors) and headings (font and text colors) are not reflecting in the site editor and live page. 

### Steps to reproduce

1. Activate Zoologist or Twenty Twenty Two red theme.
2. Go to the site editor and click Global Styles.
3. Customize a button and heading. Notice that some customizations are visible in the editor, like background color for headings and some are not. 
4. Save changes and compare the editor with the live page. 

### What you expected to happen

Customization changes to buttons and headings are both reflected in the editor and live page. 

### What actually happened

Not all customization changes to buttons and headings are reflected in the editor and live page. 

### Context

Self-testing. 

### Zoologist button in the site editor. The changes in font is reflected in the editor but not the size and colors: 

<img width="1206" alt="Screen Shot 2022-10-11 at 4 59 51 PM" src="https://user-images.githubusercontent.com/36863633/195130849-cbdf56a7-9767-40fc-8893-088030eb27f9.png">

<img width="1208" alt="Screen Shot 2022-10-11 at 5 00 06 PM" src="https://user-images.githubusercontent.com/36863633/195130946-dd66b653-1ef3-47da-9664-036b3db182ac.png">

### Zoologist button in the  live page, also notice the heading: 

<img width="750" alt="Screen Shot 2022-10-11 at 5 01 09 PM" src="https://user-images.githubusercontent.com/36863633/195131032-913490e4-2fd1-47f3-9c39-518efd57c857.png">

---

### Twenty Twenty Two Red heading in the site editor, only the background color is reflected: 

<img width="1148" alt="Screen Shot 2022-10-11 at 5 03 19 PM" src="https://user-images.githubusercontent.com/36863633/195131174-a5661f9a-f12c-4b80-95c0-0b7778127b76.png">

### Twenty Twenty Two Red heading in the live page, only the background color is reflected: 

<img width="843" alt="Screen Shot 2022-10-11 at 5 03 25 PM" src="https://user-images.githubusercontent.com/36863633/195131273-b54dd30d-100d-4e74-be74-3687fe21dfd4.png">

### Platform (Simple, Atomic, or both?)

Simple, Atomic

### Theme-specific issue?

I've only tested Zoologist and Twenty Twenty Two Red themes. 

### Browser, operating system and other notes

Firefox 105, macOS 12.6. 

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Yes, easy to implement

### Workaround details

Use another theme or customize the heading and button in the editor  settings (not global styles) or use CSS.

**Comment by Joshgrygiel** [comment]
📌 SCRUBBING
- Tested on P2  ❌
- Tested on Simple ✅  
- Tested on AT ✅ 
- Tested on Self-hosted ✅  
- Replicable on Core - Yes

📌 FINDINGS/SCREENSHOTS/VIDEO
- After triaging on Zoologist on Simple/AT/Self Hosted I tried changing up the typography and style of both a button and a heading. I found that adding those blocks that have been altered through Global Styles directly to the page renders the blocks properly. However, if those blocks are added into a Query Loop Block, slight variations begin to appear:

![Annotation on 2022-10-11 at 12-23-05](https://user-images.githubusercontent.com/88345647/195159936-a24f5e1f-7893-4764-856f-37bbc1c124d6.png)

The top heading and button are how they should be rendered out based on the settings set in the Global Styles. Within the Query Loop Block specifically, an underline was automatically added to the button, and the heading is not utilizing the global styles set for it. 

![Screenshot on 2022-10-11 at 12-34-19](https://user-images.githubusercontent.com/88345647/195160752-2a9b6909-3af7-403c-b39b-56b4d8c005f0.png)

In the editor, you can see the difference in heading typography but the underline on the second button does not appear.

I was not able to reproduce the same results on Twenty Twenty Two Red, the blocks outside of the query loop look exactly the same as the blocks added into it:

![Screenshot on 2022-10-11 at 12-38-47](https://user-images.githubusercontent.com/88345647/195161618-74463da7-f09e-4aac-9fdd-af297ceef1cb.png)


📌 ACTIONS
- Await author feedback

📌 Message to Author
- @kspilarski Could I have you confirm on your end that buttons and heading blocks render properly when added directly to a page in the editor specifically on Zoologist, and begin to show styling discrepancies when those blocks are added inside of a Query Loop Block? I was unable to reproduce the block styles not working with blocks in general - I only noticed issues when the blocks were added into a query loop block. I was unable to reproduce at all Twenty Twenty Two Red. Thank you!

**Comment by Copons** [comment]
Hi folks!
I'm fairly sure this is either a Core or theme issue, but it's very complicated to reason about.

In short, it boils down to the C in CSS: "cascading".
Styles are applied in a specific order and with specific... specificities.

With GS, we have multiple CSS sources for every element of the page:
- The Editor's own default styles.
- The theme's default styles.
- The theme.json granular (e.g. down to block types, or even some HTML element) styles.
- The user's site-wide GS customizations.
- The user's block type GS customizations (e.g. Paragraph, Image, Gallery, etc.).
- The user's HTML element GS customizations (e.g. headings and buttons).
- The user's individual block styles.

Ideally, those are registered in order, so that the browser would use the last, most-specific, style. And in most cases, this happens correctly; not in this case, though!

---

I did a quick test with Zoologist and a Button block.
I went to Global Styles -> Colors -> Buttons (which according to Gutenberg: "Set the default colors used for buttons across the site.", so it's not about a block or block type), and set the background to red.
It didn't work: the Button block stayed black.
Inspecting its source, I could see that [Zoologist's theme.json's `settings.custom.button.colors.background`](https://github.com/Automattic/themes/blob/bb88eb5012a421ba906750b48deea4d2c33ef94e/zoologist/theme.json#L63) (set to black) would override my customization.

<img width="823" alt="Screenshot 2022-10-12 at 15 49 52" src="https://user-images.githubusercontent.com/2070010/195375518-6f1243e7-a65a-4132-993b-7b54dfbe141a.png">

In this case, the cause is that, when parsing all the styles, Gutenberg assigns a higher specificity to the theme.json ones (`.wp-block-button .wp-block-button__link`) than the GS ones (`.wp-element-button, .wp-block-button__link`).

If I customize the color of that individual button (so, not in GS, but in the block's own settings), and set it to purple, it works because the style has a higher specificity (an `!important` declaration even!):

<img width="814" alt="Screenshot 2022-10-12 at 15 53 16" src="https://user-images.githubusercontent.com/2070010/195376666-527c17b3-3e0c-4db8-8454-b754cc8424ed.png">

---

The tl;dr is that this really isn't an issue related to the current GS projects on WordPress.com. 😄 

There's a chance I misunderstood the intended CSS specificity and Gutenberg's parsing is correct.
Although, I believe that the user's own GS customizations should always override the theme's styles, so IMHO here Gutenberg is at fault! 😄

**Comment by annezazu** [comment]
Possibly related: https://github.com/WordPress/twentytwentythree/issues/278

**Comment by vajrasar** [comment]
There is a contextual conversation here pdtkmj-EG-p2#comment-1121

**Comment by Copons** [comment]
To summarize, this issue is in fact a "working as intended", but very confusing from a UX standpoint.
I've started a conversation in Core Gutenberg: https://github.com/WordPress/gutenberg/issues/44931

I'm going to close this as not specifically related to Dotcom.

Thank you again for testing and reporting! 🙇‍♂️
